typedmodels/models.py

```diff
@@ -305,20 +305,32 @@ def __init__(self, *args, **kwargs):
             self.recast()
 
     def recast(self, typ=None):
-        if not self.type:
-            if not hasattr(self, '_typedmodels_type'):
-                # Ideally we'd raise an error here, but the django admin likes to call
-                # model() and doesn't expect an error.
-                # Instead, we raise an error when the object is saved.
-                return
-            self.type = self._typedmodels_type
-
         for base in self.__class__.mro():
             if issubclass(base, TypedModel) and hasattr(base, '_typedmodels_registry'):
                 break
         else:
             raise ValueError("No suitable base class found to recast!")
 
+        if not self.type:
+            if not hasattr(self, '_typedmodels_type'):
+                # This is an instance of an untyped model
+                if typ is None:
+                    # recast() is probably being called by __init__() here.
+                    # Ideally we'd raise an error here, but the django admin likes to call
+                    # model() and doesn't expect an error.
+                    # Instead, we raise an error when the object is saved.
+                    return
+                else:
```

Comment:
You don't need this `else` block. As it stands, you'll assign `type`, then on (new) line 339 re-assign `typ`, then on (new) line 346 re-assign `type` from `typ`.

Without the `else` block, (new) line 337 with (new) line 342 handles everything correctly in the `animal.recast(Feline)` use case.

Reply:
ah, good point! I'll nuke this whole chunk 😄